### ISSUE TITLE
Failing to get encoders for connector

### ISSUE THREAD

**Issue by ids1024** [Smithay/drm-rs#123]
```rust
let card = Card::open_global();
let resources = card.resource_handles().unwrap();

for (i, handle) in resources.connectors().iter().enumerate() {
    let info = card.get_connector(*handle).unwrap();
    eprintln!("Connector {} encoders: {:?}", i, info.encoders());
}
```

```
Connector 0 encoders: [Some(encoder::Handle(77)), None, None]
Connector 1 encoders: [Some(encoder::Handle(83)), None, None]
Connector 2 encoders: [Some(encoder::Handle(88)), None, None]
Connector 3 encoders: [Some(encoder::Handle(93)), None, None]
Connector 4 encoders: [None, None, None]
Connector 5 encoders: [None, None, None]
```

`drm_info` shows `Encoders: {4, 5, 6, 7}` for Connector 4 and Connector 5. Which are DisplayPort over USB-C devices connected to my HP Dev One laptop with AMD graphics.

Not sure exactly what is wrong with the ioctl call, but it seems assuming there would be at most 3 encoders is not correct.

**Comment by ids1024** [comment]
Using an array that fits 4 connectors instead of 3 seems to fix this.

Edit: The Linux kernel doesn't write encoders unless `out_resp->count_encoders >= encoders_count`. But what is the maximum `encoders_count`, if not 3?

**Comment by ids1024** [comment]
Looking at the code in the kernel and the implementation in libdrm (`_drmModeGetConnector`), it seems like the correct way to is to call `DRM_IOCTL_MODE_GETCONNECTOR` with the counts set to 0 to get the lengths needed, then again with appropriately sized arrays. Then continue calling in a loop if the counts are different, since there could be a hotplug in between the ioctl calls.

This is probably needed in other places in drm-rs too. Fun.

**Comment by Drakulix** [comment]
I have addressed parts of this problem for modes already here: https://github.com/Smithay/drm-rs/pull/121

I'll make sure to give encoders the same treatment.

**Comment by Drakulix** [comment]
Will hopefully be fixed by #124
